pkg/grpc/universalapi/api_actors.go

```diff
@@ -47,3 +53,51 @@ func (a *UniversalAPI) WaitForActorsReady(ctx context.Context) {
 	case <-a.actorsReadyCh:
 	}
 }
+
+// This function makes sure that the actor subsystem is ready.
+func (a *UniversalAPI) actorReadinessCheck(ctx context.Context) error {
+	a.WaitForActorsReady(ctx)
+
+	if a.Actors == nil {
+		a.Logger.Debug(messages.ErrActorRuntimeNotFound)
+		return messages.ErrActorRuntimeNotFound
+	}
+
+	return nil
+}
+
+func (a *UniversalAPI) DeleteActorState(ctx context.Context, in *runtimev1pb.DeleteActorStateRequest) (*emptypb.Empty, error) {
+	if err := a.actorReadinessCheck(ctx); err != nil {
+		return &emptypb.Empty{}, err
+	}
+
+	actorType := in.ActorType
+	actorID := in.ActorId
+	key := in.Key
+
+	hosted := a.Actors.IsActorHosted(ctx, &actors.ActorHostedRequest{
```

Comment:
This may need further discussion. By doing this check, you're essentially making it so only an actor can delete its own state? It may be what we want, but then it requires an actor to implement the logic to clean up after itself.

I looked at the issue that was linked and couldn't find details on how this API was meant to be used. Deciding on how this API will be used may be key first